### PR TITLE
fix: fix various analytics issues

### PR DESCRIPTION
**Pull request by khennes**

### CODE REVIEW

File: docker-compose.override.dev.yml [hunk 1]
```diff
@@ -4,7 +4,6 @@ services:
     environment:
       - NODE_ENV=development
```
Comment (lucasmarshall) [inline docker-compose.override.dev.yml:5]: Probably don't want to check this in

**Review by lucasmarshall — CHANGES_REQUESTED** [review]
Need to keep the analytics disabled for the dev override

File: apps/api/index.ts [hunk 2]
```diff
@@ -45,6 +45,8 @@ app.use(
   })
 );
 
+app.use(posthogMiddleware);
```
Comment (lucasmarshall) [inline apps/api/index.ts:48]: Doesn't this log success always, even for errors?

**Comment by khennes** [comment]
> Need to keep the analytics disabled for the dev override

Why do we want that?

File: apps/api/index.ts (hunk 2 above)
Comment (khennes) [reply apps/api/index.ts:48]: Yes, it's more of an "attempt" than a success. I couldn't figure out how to emit an event for a successful request.

**Comment by khennes** [comment]
> > Need to keep the analytics disabled for the dev override
> 
> Why do we want that?

For posterity: we'll keep analytics enabled for the dev override so that we have visibility into external folks who want to develop against supaglue.

File: apps/api/index.ts (hunk 2 above)
Comment (khennes) [reply apps/api/index.ts:48]: We'll rename the result type to `attempt` for now and figure out the middleware later.

**Pull request merged**